integrations_test.go

```diff
@@ -1,17 +1,66 @@
 package rockset_test
 
 import (
+	"context"
 	"errors"
 	"log"
+	"net/http"
 	"os"
+	"reflect"
 	"testing"
+	"unsafe"
 
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	"github.com/rockset/rockset-go-client"
+	"github.com/rockset/rockset-go-client/openapi"
 	"github.com/rockset/rockset-go-client/option"
 )
 
+func (s *IntegrationsSuite) TestCreateAzureBlob() {
+	name := "azureBlobTest"
+	connectionString := ""
+	ctx := testCtx()
+
+	integrationRequest := openapi.ApiCreateIntegrationRequest{
+		ApiService: s.IntegrationsApi,
+	}
+	InjectCtx(ctx, &integrationRequest)
+	integrationRequest.ApiService = s.IntegrationsApi
```

Comment:
Does this work? I just eyeballing based on https://github.com/rockset/rockset-go-client/blob/f85b9a8ec9b2d64c564d31dcffb9463e8d20f79f/openapi/api_integrations.go#L115-L120 , but if so, you can avoid the reflection.

```suggestion
	integrationRequest := s.IntegrationsApi.CreateIntegration(ctx)
```

Reply:
It works because we have mocked the `CreateIntegration` call.
However, `ctx` is not accessible outside of the `openapi` package.

The issue is compounded further because later function calls, nested deep into `openapi` use that value, assuming it has been set by the call.

This is why it needs `reflect` and `unsafe` to bypass Go's type safety.

Reply:
Oh I see. the mocking interferes. I'll take a closer look.

Reply:
If what @notnoop's suggestion doesn't work, why not modify the openapi generated code to expose the context or add a method to let you set the context?

Reply:
Because the code isn't in our control.
It will be overwritten as soon as we add anything to the API Spec.

Reply:
I see. How about:

```go
       integrationRequest := (&openapi.IntegrationsApiService{}).CreateIntegration(ctx)
       integrationRequest.ApiService = s.IntegrationsApi
```

Reply:
`IntegrationsApiService` is an interface, not a struct.
We need an object that implements `CreateIntegration`, because the client uses that object to call `CreateIntegration`

https://github.com/rockset/rockset-go-client/blob/clo/sre-1601-update-go-client-to-allow-for-unit/integrations.go#L60

And the NewClient just creates one for itself.
https://github.com/rockset/rockset-go-client/blob/master/rockset.go#L93

Hence we need to set the `RockClient.ApiService` to our own mock service.
https://github.com/rockset/rockset-go-client/blob/clo/sre-1601-update-go-client-to-allow-for-unit/integrations_test.go#L53

Reply:
And we want to Mock `CreateIntegration` because it is outside the code we are testing.

Reply:
I off course meant, write some code to automatically modify the generated code, after we run the generator - go code can easily be parsed and modified with the parser [package](https://pkg.go.dev/go/parser).

You can also modify the generator.

Reply:
Why do you feel the need to go through all that trouble to mock what is basically a constructor with no side-effects? What is it that you gain from all that added complexity?

To me it is overly pedantic, which results in a lot of bloat for no real gain.

When you work with code that you have little control over, you can't be as nit-picky as if you have 100% control over it. If we owned the generator code, we would change it so it would be easier to test, but we don't, so we should take a more pragmatic approach to it.

Adding unit tests is good, but not at the expense of using `unsafe.Pointer()` and reflection.

Reply:
try something like this instead:
```
func TestRockClient_CreateGCSIntegration(t *testing.T) {
	ctx := testCtx()
	rc, err := rockset.NewClient()
	require.NoError(t, err)

	rc.IntegrationsApi = &fakeAPI{api: rc.IntegrationsApi}

	_, err = rc.CreateGCSIntegration(ctx, "name", "json",
		option.WithGCSIntegrationDescription("created by rockset integration tests"))
	require.NoError(t, err)

}

type fakeAPI struct {
	openapi.IntegrationsApi
	api openapi.IntegrationsApi
}

func (f *fakeAPI) CreateIntegration(ctx context.Context) openapi.ApiCreateIntegrationRequest {
	return f.api.CreateIntegration(ctx)
}
```